include/tunable_reduction.h

```diff
@@ -380,21 +386,36 @@ namespace quda
       do {
         rtn = Tunable::advanceBlockDim(param);
       } while (rtn && !is_power2(param.block.x));
-      return rtn;
+
+      if (rtn) {
+        return true;
+      } else {
+        if (param.block.z < n_batch && param.block.z < device::max_threads_per_block_dim(2) &&
+            param.block.x * param.block.y * (param.block.z + 1) <= device::max_threads_per_block() &&
+            param.block.z < n_batch_block_max) {
+          param.block.z++;
+          param.grid.z = (n_batch + param.block.z - 1) / param.block.z;
+          return true;
+        } else { // we have run off the end so let's reset
+          param.block.z = 1;
+          param.grid.z = (n_batch + param.block.z - 1) / param.block.z;
+          return false;
+        }
+      }
     }
 
     void initTuneParam(TuneParam &param) const
     {
       Tunable::initTuneParam(param);
-      param.block.y = 1;
-      param.grid.y = n_batch;
+      param.block = {param.block.x, 1, 1};
```

Comment:
One way to decrease the tuning time is to start not from `block.z == 1` but from some other larger user-defined number. But this is definitely for a future PR.